matrix-connection.c

```diff
@@ -237,22 +238,37 @@ void matrix_connection_start_login(PurpleConnection *pc)
 }
 
 
-static void _join_completed(MatrixConnectionData *conn,
+static void _tell_purple_joined_callback(MatrixConnectionData *conn,
         gpointer user_data,
         JsonNode *json_root,
         const char *raw_body, size_t raw_body_len, const char *content_type)
 {
-    GHashTable *components = user_data;
     JsonObject *root_obj;
     const gchar *room_id;
 
     root_obj = matrix_json_node_get_object(json_root);
     room_id = matrix_json_object_get_string_member(root_obj, "room_id");
-    purple_debug_info("matrixprpl", "join %s completed", room_id);
-
-    g_hash_table_destroy(components);
+    if (room_id && *room_id) {
+        matrix_room_join_conversation(conn->pc, room_id);
```

Comment:
This should be unnecessary, because the response to `/sync` will tell us we have joined the room. Why is it necessary?